packages/eas-cli/src/submissions/archive-source/ArchiveTypeSource.ts

```diff
@@ -103,9 +107,11 @@ type ArchiveInferredType = ArchiveType | null;
 
 function inferArchiveTypeFromLocation(location: string): ArchiveInferredType {
   if (location.endsWith('.apk')) {
-    return ArchiveType.apk;
+    return AndroidArchiveType.apk;
   } else if (location.endsWith('.aab')) {
-    return ArchiveType.aab;
+    return AndroidArchiveType.aab;
+  } else if (location.endsWith('.ipa')) {
```

Comment:
I'm not sure if this logic is correct. What if someone was running `eas build:submit --platform android` and named the archive `my-android-app.ipa`? I know this is a stupid and infrequent case, but in this case, the result of this function is incorrect.

Reply:
Right, I should make the infer method smarter.